Fix flaky GaugeTest.testTimer test

From: https://github.com/prometheus/client_java/actions/runs/23299167470/job/67754952796?pr=1967

```
Error:  Tests run: 12, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.396 s <<< FAILURE! -- in io.prometheus.metrics.core.metrics.GaugeTest
Error:  io.prometheus.metrics.core.metrics.GaugeTest.testTimer -- Time elapsed: 0.034 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  0.022711679
to be close to:
  0.012
by less than 0.005 but difference was 0.010711679.
(a difference of exactly 0.005 being considered valid)
	at io.prometheus.metrics.core.metrics.GaugeTest.testTimer(GaugeTest.java:91)
```

--- a/prometheus-metrics-core/src/test/java/io/prometheus/metrics/core/metrics/GaugeTest.java
+++ b/prometheus-metrics-core/src/test/java/io/prometheus/metrics/core/metrics/GaugeTest.java
@@ -82,12 +82,11 @@ class GaugeTest {
   }
 
   @Test
-  @SuppressWarnings("try")
-  public void testTimer() throws InterruptedException {
-    try (Timer ignored = noLabels.startTimer()) {
-      Thread.sleep(12);
-    }
-    assertThat(getValue(noLabels)).isGreaterThan(0.01);
+  void testTimer() {
+    Timer timer = noLabels.startTimer();
+    double duration = timer.observeDuration();
+    assertThat(duration).isPositive();
+    assertThat(getValue(noLabels)).isEqualTo(duration);
   }
 
   @Test

